File: CPlus/lengthoflongestsubstring.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int left=-1;
        int longest=0;
        unordered_map<int,int> map;
        
        for(int i=0;i<s.size();i++){
            if(map.count(s[i]) && map[s[i]] > left){
                left=map[s[i]];
            }
            map[s[i]]=i;
            longest=max(longest,i-left);            
        }
        return longest;
    }
};

File: Python/lengthOfLongestSubstring.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        if not s or len(s)==0:
            return 0
        
        start=0
        
        longest=0
        hashmap={}
        
        for i in range(len(s)):
            
            if s[i] in hashmap:
                start=max(start,hashmap[s[i]]+1)
            longest=max(i-start+1,longest)
            hashmap[s[i]]=i
        
        
        return longest